### PR TITLE
Working holistic from ovms

### PR DESCRIPTION
**Pull request by rasapala**

### CODE REVIEW

**Review by dtrawins — APPROVED** [review]

File: mediapipe/calculators/ovms/modelapiovmsinferencecalculator.cc [hunk 1]
```diff
@@ -93,10 +159,98 @@ static Tensor convertOVTensor2MPTensor(const ov::Tensor& t) {
...
+        return Precision::UNDEFINED;
+    }
+    return it->second;
+}
+
 static tensorflow::Tensor convertOVTensor2TFTensor(const ov::Tensor& t) {
     using tensorflow::Tensor;
     using tensorflow::TensorShape;
-    auto datatype = ovms::getPrecisionAsDataType(ovms::ovElementTypeToOvmsPrecision(t.get_element_type()));
+    auto datatype = getPrecisionAsDataType(ovElementTypeToOvmsPrecision(t.get_element_type()));
```
Comment (atobiszei) [inline mediapipe/calculators/ovms/modelapiovmsinferencecalculator.cc]: Here don't do two step conversion - add OV datatype to TF datatype

File: mediapipe/calculators/ovms/modelapiovmsinferencecalculator.cc [hunk 2]
```diff
@@ -93,10 +159,98 @@ static Tensor convertOVTensor2MPTensor(const ov::Tensor& t) {
...
+*/
+    };
+    auto it = precisionMap.find(type);
+    if (it == precisionMap.end()) {
+        return Precision::UNDEFINED;
+    }
+    return it->second;
+}
+
+ov::element::Type_t ovmsPrecisionToIE2Precision(Precision precision) {
```
Comment (atobiszei) [inline mediapipe/calculators/ovms/modelapiovmsinferencecalculator.cc]: To remove

File: mediapipe/calculators/ovms/modelapiovmsinferencecalculator.cc [hunk 3]
```diff
@@ -93,10 +159,98 @@ static Tensor convertOVTensor2MPTensor(const ov::Tensor& t) {
     return outputTensor;
 }
 
+Precision ovElementTypeToOvmsPrecision(ov::element::Type_t type) {
```
Comment (atobiszei) [inline mediapipe/calculators/ovms/modelapiovmsinferencecalculator.cc]: To remove

File: mediapipe/calculators/ovms/modelapiovmsinferencecalculator.cc [hunk 4]
```diff
@@ -67,6 +66,73 @@ const std::string MPTENSORS_TAG{"TENSORS"};
...
+    }
+    bool allOf = std::all_of(str.begin(),
+        std::next(str.begin(), prefix.size()),
+        [&it](const char& c) {
+            return c == *(it++);
+        });
+    return allOf;
+}
+
+TFSDataType getPrecisionAsDataType(Precision precision) {
```
Comment (atobiszei) [inline mediapipe/calculators/ovms/modelapiovmsinferencecalculator.cc]: To remove

File: mediapipe/calculators/ovms/modelapiovmsinferencecalculator.cc [hunk 5]
```diff
@@ -67,6 +66,73 @@ const std::string MPTENSORS_TAG{"TENSORS"};
 const std::string TFLITE_TENSOR_TAG{"TFLITE_TENSOR"};
 const std::string TFLITE_TENSORS_TAG{"TFLITE_TENSORS"};
 
+enum class Precision {
```
Comment (atobiszei) [inline mediapipe/calculators/ovms/modelapiovmsinferencecalculator.cc]: This shouldn;t be needed,

File: mediapipe/calculators/ovms/modelapiovmsinferencecalculator.cc [hunk 6]
```diff
@@ -114,7 +268,7 @@ static tensorflow::Tensor convertOVTensor2TFTensor(const ov::Tensor& t) {
 
 static ov::Tensor convertTFTensor2OVTensor(const tensorflow::Tensor& t) {
     void* data = t.data();
-    auto datatype = ovms::ovmsPrecisionToIE2Precision(ovms::TFSPrecisionToOvmsPrecision(t.dtype()));
+    auto datatype = ovmsPrecisionToIE2Precision(TFSPrecisionToOvmsPrecision(t.dtype()));
```
Comment (atobiszei) [inline mediapipe/calculators/ovms/modelapiovmsinferencecalculator.cc]: Here as well avoid double step conversion

File: mediapipe/calculators/ovms/modelapiovmsinferencecalculator.cc (hunk 1 above)
Comment (rasapala) [reply mediapipe/calculators/ovms/modelapiovmsinferencecalculator.cc]: Done.

File: mediapipe/calculators/ovms/modelapiovmsinferencecalculator.cc (hunk 2 above)
Comment (rasapala) [reply mediapipe/calculators/ovms/modelapiovmsinferencecalculator.cc]: Done.

File: mediapipe/calculators/ovms/modelapiovmsinferencecalculator.cc (hunk 3 above)
Comment (rasapala) [reply mediapipe/calculators/ovms/modelapiovmsinferencecalculator.cc]: Done.

File: mediapipe/calculators/ovms/modelapiovmsinferencecalculator.cc (hunk 4 above)
Comment (rasapala) [reply mediapipe/calculators/ovms/modelapiovmsinferencecalculator.cc]: Done.

File: mediapipe/calculators/ovms/modelapiovmsinferencecalculator.cc (hunk 5 above)
Comment (rasapala) [reply mediapipe/calculators/ovms/modelapiovmsinferencecalculator.cc]: Removed.

File: mediapipe/calculators/ovms/modelapiovmsinferencecalculator.cc (hunk 6 above)
Comment (rasapala) [reply mediapipe/calculators/ovms/modelapiovmsinferencecalculator.cc]: Done.

File: mediapipe/calculators/ovms/modelapiovmsinferencecalculator.cc [hunk 7]
```diff
@@ -67,6 +66,48 @@ const std::string MPTENSORS_TAG{"TENSORS"};
...
+        {ov::element::Type_t::i64, TFSDataType::DT_INT64},
+        {ov::element::Type_t::i32, TFSDataType::DT_INT32},
+        {ov::element::Type_t::i16, TFSDataType::DT_INT16},
+        {ov::element::Type_t::i8, TFSDataType::DT_INT8},
+        {ov::element::Type_t::u64, TFSDataType::DT_UINT64},
+        {ov::element::Type_t::u16, TFSDataType::DT_UINT16},
+        {ov::element::Type_t::u8, TFSDataType::DT_UINT8},
+        //    {Precision::MIXED, TFSDataType::DT_INVALID},
+        //    {Precision::Q78, TFSDataType::DT_INVALID},
+        //    {Precision::BIN, TFSDataType::DT_INVALID},
```
Comment (atobiszei) [inline mediapipe/calculators/ovms/modelapiovmsinferencecalculator.cc:100]: Remove

File: mediapipe/calculators/ovms/modelapiovmsinferencecalculator.cc [hunk 8]
```diff
@@ -67,6 +66,48 @@ const std::string MPTENSORS_TAG{"TENSORS"};
...
+        {ov::element::Type_t::boolean, TFSDataType::DT_BOOL}
+        //    {Precision::CUSTOM, TFSDataType::DT_INVALID}
```
Comment (atobiszei) [inline mediapipe/calculators/ovms/modelapiovmsinferencecalculator.cc:102]: Remove
```suggestion
```

File: mediapipe/calculators/ovms/modelapiovmsinferencecalculator.cc (hunk 7 above)
Comment (atobiszei) [inline mediapipe/calculators/ovms/modelapiovmsinferencecalculator.cc]:
```suggestion
```

File: mediapipe/calculators/ovms/modelapiovmsinferencecalculator.cc (hunk 8 above)
Comment (atobiszei) [inline mediapipe/calculators/ovms/modelapiovmsinferencecalculator.cc]:
```suggestion
```

**Review by atobiszei — APPROVED** [review]

**Pull request merged**